server/scsynth/SC_PulseAudio.cpp

```diff
@@ -0,0 +1,320 @@
+        int64 oscInc = mOSCincrement = (uint64)((m_DLL.Period() / numBufs) * kSecondsToOSCunits + .5);
+        mSmoothSampleRate = m_DLL.SampleRate();
+        double oscToSamples = mOSCtoSamples = mSmoothSampleRate * kOSCtoSecs /* 1/pow(2,32) */;
+
+        // main loop
+        for (int i = 0; i < numBufs; ++i, mWorld->mBufCounter++, bufFramePos += bufFrames) {
+            int32 bufCounter = mWorld->mBufCounter;
+            int32* tch;
+
+            // Process inputs, marking them as touched
+            tch = inTouched;
+            for (int k = 0; k < m_inputChannelCount; ++k) {
+                *tch++ = bufCounter;
+                float* dst = inBuses + k * bufFrames;
+                const float* src = (float*)inputBuffer + k + bufFramePos;
+                for (int n = 0; n < bufFrames; n++) {
+                    *dst = *src;
+                    src += m_outputChannelCount;
+                    dst++;
+                }
+            }
+
+            // run engine
+            int64 schedTime;
+            int64 nextTime = oscTime + oscInc;
+            while ((schedTime = mScheduler.NextTime()) <= nextTime) {
+                float diffTime = (float)(schedTime - oscTime) * oscToSamples + 0.5;
+                float diffTimeFloor = floor(diffTime);
+                world->mSampleOffset = (int)diffTimeFloor;
+                world->mSubsampleOffset = diffTime - diffTimeFloor;
+
+                if (world->mSampleOffset < 0)
+                    world->mSampleOffset = 0;
+                else if (world->mSampleOffset >= world->mBufLength)
+                    world->mSampleOffset = world->mBufLength - 1;
+
+                SC_ScheduledEvent event = mScheduler.Remove();
+                event.Perform();
+            }
+            world->mSampleOffset = 0;
+            world->mSubsampleOffset = 0.0f;
+
+            World_Run(world);
+
+            // Process outputs (considering which ones have been touched)
+            tch = outTouched;
+            for (int k = 0; k < m_outputChannelCount; ++k) {
+                float* dst = (float*)outputBuffer + k + bufFramePos;
+                if (tch[k] == bufCounter) {
+                    const float* src = outBuses + k * bufFrames;
+                    for (int n = 0; n < bufFrames; n++) {
+                        *dst = *src;
+                        src++;
+                        dst += m_outputChannelCount;
+                    }
+                } else {
+                    *dst = 0;
+                    dst += m_outputChannelCount;
+                }
+            }
+            // update buffer time
+            oscTime = mOSCbuftime = nextTime;
+        }
+    } catch (std::exception& exc) {
+        scprintf("SC_PulseAudioDriver: exception in real time: %s\n", exc.what());
+    } catch (...) {
+        scprintf("SC_PulseAudioDriver: unknown exception in real time\n");
+    }
+#if 0
```

Comment:
I can see why this is unfinished, but this should be fixed before merging -- not only for users but also for benchmarking during testing.

If there's no analog to Pa_GetStreamCpuLoad(), we could calculate how much time it took to process the audio divided by the duration of the block being calculated, times 100.0.

Reply:
Will get to this soon.

Reply:
I have added this in the latest commit